include/boost/property_tree/detail/info_parser_read.hpp

```diff
@@ -210,7 +212,13 @@ namespace boost { namespace property_tree { namespace info_parser
         std::stack<Ptree *> stack;
         stack.push(&pt);                // Push root ptree on stack initially
 
-        try {
+        // When compiling without exception support there is no formal
+        // parameter "e" in the catch handler.  Declaring a local variable
+        // here does not hurt and will be "used" to make the code in the
+        // handler  compilable although the code will never be executed.
+        info_parser_error e("", "", 0); ignore_unused(e);
```

Comment:
I actually don't like this hack because it *could* generate code. I'm going to email the mailing list to come up with an alternative.